File: algorithms/math/python/lcm.py
import math

from algorithms.math.python.gcd import _gcd_manual


def lcm(a: int, b: int) -> int:
    return math.lcm(a, b)


def _lcm_manual(a: int, b: int) -> int:
    if a == 0 or b == 0:
        return 0
    absA = abs(a)
    absB = abs(b)
    return (absA // _gcd_manual(absA, absB)) * absB

File: algorithms/math/javascript/lcm.js
function lcm(a, b) {
  if (!Number.isInteger(a) || !Number.isInteger(b)) {
    throw new TypeError(
      `lcm() arguments must be integers, got: a=${typeof a}, b=${typeof b}`
    );
  }
  if (a === 0 || b === 0) {
    return 0n;
  }
  const absA = BigInt(Math.abs(a));
  const absB = BigInt(Math.abs(b));
  const gcd = (x, y) => {
    while (y !== 0n) {
      [x, y] = [y, x % y];
    }
    return x;
  };
  return (absA / gcd(absA, absB)) * absB;
}
